tests/test_data_loader.py

```diff
@@ -396,3 +398,16 @@ def test_end_of_dataloader_dispatcher(self):
         # Test it also works on the second iteration
         for idx, _ in enumerate(dataloader):
             assert dataloader.end_of_dataloader == (idx == 3)
+
+    def test_seedable_random_sampler_is_fully_applied(self):
+        Accelerator()
+        dataloader = DataLoader(list(range(10)), batch_size=1, shuffle=True)
+        prepared_data_loader = prepare_data_loader(
+            dataloader=dataloader,
+            num_processes=2,
+            process_index=0,
+            use_seedable_sampler=True,
+        )
+        batch_sampler = prepared_data_loader.batch_sampler
+        assert isinstance(batch_sampler, BatchSamplerShard)
+        assert isinstance(batch_sampler.batch_sampler.sampler, SeedableRandomSampler)
```

Comment:
This should generally be under `src/accelerate/test_utils/scripts/test_script.py`, you'll find similar tests in there for `prepare_data_loader` you can mimic.

Let's move that there please?

Reply:
Yeah of course! I'll move it now.